internal/deepdoc/native/validate_proto.py

```
@@ -0,0 +1,95 @@
+        p = np.array(p, float)
+        return 0.5 * abs(np.dot(p[:, 0], np.roll(p[:, 1], -1)) -
+                         np.dot(p[:, 1], np.roll(p[:, 0], -1)))
+
+    def inter(p, q):
+        cp = [tuple(x) for x in p]
+        for i in range(len(q)):
+            a1, b1 = np.array(q[i]), np.array(q[(i + 1) % len(q)])
+            e = (b1[1] - a1[1], a1[0] - b1[0], b1[0] * a1[1] - a1[0] * b1[1])
+            ncp = []
+            for j in range(len(cp)):
+                cur = np.array(cp[j]); nxt = np.array(cp[(j + 1) % len(cp)])
```

Ruff (0.16.2)

[error] Multiple statements on one line (semicolon)

(E702)